src/app/content/routes.ts

```diff
@@ -1,15 +1,36 @@
+import omit from 'lodash/fp/omit';
 import pathToRegexp from 'path-to-regexp';
 import Loadable from 'react-loadable';
 import { Route } from '../navigation/types';
 import { getUrlRegexParams } from '../navigation/utils';
 import { SelectedResult } from './search/types';
 import { Params } from './types';
+import { findPathForParams } from './utils/urlUtils';
 
 const MATCH_UUID = '[\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12}';
+const base = '/books/:book/pages/:page';
 
-const CONTENT_PATH = '/books/:book_slug/pages/:page_slug';
-const UUID_CONTENT_PATH = `/books/:book_uuid(${MATCH_UUID})@:book_version/pages/:page_slug`;
-const VERSIONED_CONTENT_PATH = '/books/:book_slug@:book_version/pages/:page_slug';
+/*
+ * Recursively creates combinations of supplied replacements
+ * for the base parameter in an url
+ */
+
+const injectParamsToBaseUrl = (baseUrl: string, params: {[key: string]: string[]}): string[] => {
+  const paramKeys: Array<keyof typeof params> = Object.keys(params);
+  const keyToInject = paramKeys[0];
+
+  return params[keyToInject].reduce((output, value) => {
+    const injected = baseUrl.replace(new RegExp(`(?<=:)${keyToInject}`), value);
+    return paramKeys.length === 1
+      ? [...output, injected]
+      : [...output, ...injectParamsToBaseUrl(injected, omit([keyToInject], params))];
+  }, [] as string[]);
+};
```

Comment:
I'm not sure how to make it work if it's in `urlUtils` since I think the `react-loadable` makes the file load separately?

Reply:
this is probably a @TomWoodward question

Reply:
`react-loadable` isn't the problem. the problem is that there is a dependency loop between the `routes` and `urlUtils` file, everything works fine as long as the logic is in functions that only execute after everything has been mapped up, but with logic in the root of the file your modules will be `undefined`.

but this function would be better in `navigation/utils` anyway, so that should work better for you